### DISCUSSION TITLE
Is FlowForge and UI-builder compatible?

### DISCUSSION THREAD

**Discussion by Sulomus** [FlowFuse/flowfuse#1620] [Q&A] [answered]
Is there a way to make FlowForge and UI-builder work together?

I installed the contrib-ui-builder package into FlowForge cloud but can't make any changes because everytime it asks me for user and password (and I have no idea what combination to enter):

![image](https://user-images.githubusercontent.com/49107150/215059368-7961d24b-12ce-497f-86fd-dc539c24ffb0.png)

_But even if I could enter the username password I want to ask if putting any more effort into this is logical - or FlowForge and UI-builder are not yet compatible (my main concern is also if it will work OKay on the EdgeDevice)?_

**Comment by hardillb** [comment, marked as answer]
The only reason I can think of that cause a HTTP Basic Auth password prompt to show up like that is if UI Builder has bound to the `httpNodes` routes and you have enabled security under the projects settings:

![image](https://user-images.githubusercontent.com/1800845/215067249-dc43ddb8-55eb-408d-88a7-61264a8005a7.png)

Anything accesses by the editor should probably be bound to the `htttpAdmin` routes so as not to trigger the authentication (but this should still apply to access the UI builder output)

Also since UI Builder writes things to the userDir (`/data`) in the container this will be lost on a container/stack restart, while I believe it will be rebuilt as part of the flow starting up, this is not the best in a cloud environment.

**Reply by Sulomus** [reply]
Okay thank you I will try it.

> Also since UI Builder writes things to the userDir (/data) in the container this will be lost on a container/stack restart, while I believe it will be rebuilt as part of the flow starting up, this is not the best in a cloud environment.

I am only concerned if it will work on the EdgeDevice. Can you make any prediction about that?

**Reply by hardillb** [reply]
I don't use UI Builder so I have no first hand experience, but I don't see why it shouldn't work as long as it is capable of recreating all the parts it stores in `/data` (which will be `/opt/flowforge-device/project` on the device) when it loads the flow for the first time.
